x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/components/choose_connector_selectable.tsx

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { css } from '@emotion/react';
+
+import {
+  EuiBadge,
+  EuiFlexItem,
+  EuiIcon,
+  EuiInputPopover,
+  EuiSelectable,
+  EuiSelectableOption,
+  useEuiTheme,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+interface ChooseConnectorSelectableProps {
+  allConnectors: Array<{
+    description: string;
+    iconPath: string;
+    isBeta: boolean;
+    isNative: boolean;
+    isTechPreview: boolean;
+    name: string;
+  }>;
+  connectorSelected: {
+    description: string;
+    iconPath: string;
+    isBeta: boolean;
+    isNative: boolean;
+    isTechPreview: boolean;
+    name: string;
+  };
+  selfManaged: boolean;
+  setConnectorSelected: Function;
+  setSelfManaged: Function;
+}
+interface OptionData {
+  secondaryContent?: string;
+}
+
+export const ChooseConnectorSelectable: React.FC<ChooseConnectorSelectableProps> = ({
+  setConnectorSelected,
+  connectorSelected,
+  selfManaged,
+  setSelfManaged,
+  allConnectors,
+}) => {
+  const { euiTheme } = useEuiTheme();
+  const [isOpen, setIsOpen] = useState(false);
+  const [isSearching, setIsSearching] = useState(true);
+  const [selectableOptions, selectableSetOptions] = useState<
+    Array<EuiSelectableOption<OptionData>>
+  >([]);
+  const initialOptions = allConnectors.map(
+    (
```

Comment:
Changing the code for this one:

```
  const getInitialOptions = (
    allConnectors: ConnectorDefinition[],
    connectorSelected: ConnectorDefinition,
    selfManaged: boolean
  ) => {
    return allConnectors.map((connector, key) => {
      const append: JSX.Element[] = [];
      if (connector.isTechPreview) {
        append.push(
          <EuiBadge iconType="beaker" color="hollow">
            {i18n.translate(
              'xpack.enterpriseSearch.chooseConnectorSelectable.thechPreviewBadgeLabel',
              { defaultMessage: 'Tech preview' }
            )}
          </EuiBadge>
        );
      }
      if (connector.isBeta) {
        append.push(
          <EuiBadge iconType={'beta'} color="hollow">
            {i18n.translate('xpack.enterpriseSearch.chooseConnectorSelectable.BetaBadgeLabel', {
              defaultMessage: 'Beta',
            })}
          </EuiBadge>
        );
      }
      if (!selfManaged && !connector.isNative) {
        append.push(
          <EuiBadge iconType={'warning'} color="warning">
            {i18n.translate(
              'xpack.enterpriseSearch.chooseConnectorSelectable.OnlySelfManagedBadgeLabel',
              {
                defaultMessage: 'Self managed',
              }
            )}
          </EuiBadge>
        );
      }

      return {
        append,
        key: key.toString(),
        label: connector.name,
        prepend: <EuiIcon size="l" type={connector.iconPath} />,
      };
    });
  };

  const initialOptions = getInitialOptions(allConnectors, connectorSelected, selfManaged);
```

It works, but the linter is yelling because these variables are already declared as component props:
![CleanShot 2024-07-29 at 13 33 49@2x](https://github.com/user-attachments/assets/7d67c25c-210e-456d-ba3d-ac8ef8eb5a18)